src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/Namespace/SetAzureEventHubNameSpaceRegenerateKeys.cs

```diff
@@ -0,0 +1,63 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Management.EventHub.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands.Namespace
+{
+    [Cmdlet(VerbsCommon.Set, EventHubNamespaceRegenerateKeysVerb), OutputType(typeof(ResourceListKeys))]
+    public class SetAzureRmNameSpaceRegenerateKeys : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The name of the resource group")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroup { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "Namespace Name.")]
+        [ValidateNotNullOrEmpty]
+        public string NamespaceName { get; set; }
+        
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = "Authorization Rule Name.")]        
+        [ValidateNotNullOrEmpty]
+        public string AuthorizationRule { get; set; }        
+
+        [Parameter(Mandatory = true,
+            Position = 3,
+            ParameterSetName = RegenerateKeysSetName,
+            HelpMessage = "Regenerate Keys - 'PrimaryKey'/'SecondaryKey'.")]
+        [ValidateSet(RegeneKeys.PrimaryKey,
+            RegeneKeys.SecondaryKey,
+            IgnoreCase = true)]
+        [ValidateNotNullOrEmpty]        
+        public string RegenerateKeys { get; set; }
```

Comment:
See above - likely this should just be 'Key' cmdlets and parameters should not use plural names